test/aircraft/FlightManagementSystem/Fms.spec.js

```diff
@@ -819,6 +820,74 @@ ava('._prependLegCollectionWithRouteAmendment() adds LegModels for each routeStr
     t.true(fms.legCollection[1].routeString === routeAmmendmentMock[1]);
 });
 
+ava('._regenerateSidLeg() creates a new SID leg of identical route string for the currently assigned departure runway', (t) => {
+    const initialRunwayMock = airportModelFixture.getRunway('19L');
+    const nextRunwayMock = airportModelFixture.getRunway('19R');
+    const fms = buildFmsMockForDeparture();
+    const replaceDepartureProcedureSpy = sinon.spy(fms, 'replaceDepartureProcedure');
+    const sidLegIndex = fms._findLegIndexForProcedureType(PROCEDURE_TYPE.SID);
+    const oldWaypoints = fms.legCollection[sidLegIndex].waypointCollection;
+
+    t.true(_isEqual(fms.departureRunwayModel, initialRunwayMock));
```

Comment:
I'm not sure #831 - #835 are needed.  this looks like it's just verifying the current state of the fms, is that right?